xds/internal/balancer/edsbalancer/balancergroup.go

```diff
@@ -114,34 +199,69 @@ func (bg *balancerGroup) add(id internal.Locality, weight uint32, builder balanc
 		state: connectivity.Idle,
 	}
 	bg.pickerMu.Unlock()
+
+	// Store data in static map, and then check to see if bg is started.
+	bg.outgoingMu.Lock()
+	if _, ok := bg.idToBalancerConfig[id]; ok {
+		bg.outgoingMu.Unlock()
+		grpclog.Warningf("balancer group: adding a balancer with existing ID: %s", id)
+		return
+	}
+	config := &balancerConfig{
+		builder: builder,
+	}
+	bg.idToBalancerConfig[id] = config
+
+	if bg.outgoingStarted {
+		// Only start the balancer if bg is started.
+		config.start(&balancerGroupCC{
+			ClientConn: bg.cc,
+			id:         id,
+			group:      bg,
+		})
+	}
+	bg.outgoingMu.Unlock()
 }
 
 // remove removes the balancer with id from the group, and closes the balancer.
 //
 // It also removes the picker generated from this balancer from the picker
 // group. It always results in a picker update.
 func (bg *balancerGroup) remove(id internal.Locality) {
-	bg.mu.Lock()
-	// Close balancer.
-	if b, ok := bg.idToBalancer[id]; ok {
-		b.Close()
-		delete(bg.idToBalancer, id)
+	bg.outgoingMu.Lock()
+	delete(bg.idToBalancerConfig, id)
```

Comment:
You are just deleting the entry from the map here, and in the next line looking for the same entry to call `close()` on it. Am I missing something here?